### PR TITLE
GH-43588: [Python] Allow tuple for rename columns

### PR DESCRIPTION
**Pull request by 0x26res**
### Rationale for this change

Backward compatibility issue.

### What changes are included in this PR?

Allow tuple (as well as list) in `Table.rename_columns`

### Are these changes tested?

Unit tested

### Are there any user-facing changes?

This makes the API more open.
* GitHub Issue: #43588

### CODE REVIEW

File: cpp/submodules/parquet-testing
Comment (mapleFU) [inline cpp/submodules/parquet-testing:1]: why this is updated?

Comment (0x26res) [reply cpp/submodules/parquet-testing:1]: I've rolled it back. I don't know why it always gets in a bad state. Something to do with my set up.

**Review by jorisvandenbossche — APPROVED** [review]
Thanks, looks good!

One thing I am wondering is how specific we should be with those checks in general. Before 17.0 (before https://github.com/apache/arrow/pull/40645/), any iterable would have worked. With this change it is still only list and tuple. 
(I don't think we are very consistent about this throughout the code base)

**Comment by 0x26res** [comment]
> Thanks, looks good!
> 
> One thing I am wondering is how specific we should be with those checks in general. Before 17.0 (before #40645), any iterable would have worked. With this change it is still only list and tuple. (I don't think we are very consistent about this throughout the code base)

I agree. I was having a look there are a few user/API function where we look for list and we could look out for iterable. 

But it's not that simple. A `str` is sequence, so is a `dict`.

**Comment by 0x26res** [comment]
@jorisvandenbossche  / @pitrou / @mapleFU FYI I can't merge this myself as I don't have write access. Can one of you do it please? Thanks.

**Comment by jorisvandenbossche** [comment]
Merged!

> But it's not that simple. A `str` is sequence, so is a `dict`.

We could add some helper function "is_list_like" that checks for iterables but excludes str/bytes and mappings (although for example in `pa.array(..)` we actually do accept a mapping and treat it as a sequence .. so you get an array of the keys, that's probably not really intentional though).  Only, when you start doing specific checks instead of something generic, it can easily become a can of worms to decide which exact type to "support" in this checking or not ..

**Pull request merged**
